doc/reference/peripherals/rtio.rst

```diff
@@ -0,0 +1,49 @@
+.. _rtio_interface:
+
+Real-Time IO
+############
+
+The real-time io or rtio subsystem exposes an API for efficiently
+moving information to/from external i2c/spi devices such as sensors
+to/from a zephyr application in a latency and processing friendly
+way.
+
+Overview
+********
+
+RTIO provides two significant C structures and APIs for moving bytes around
+efficiently, blocks and pipes.
+
+Pipes are used to move blocks in a safe, and pollable manner from a source to
+a sink of data. Effectively a pipe is a single-input single-output queue of
+blocks.
+
+A :c:type:`struct rtio_block` provides the unit of data being transferred
+through a :c:type:`struct rtio_pipe`. These blocks contain a byte layout
+identifier provided by the source of the block, timestamps for when the block
+began and finished being used, as well as length and size in bytes.
+
+Blocks are used to allow for simple DMA transfers as well as controlling the
+latency between a source and sink.
+
+Sources can be sensors such has a high rate accelerometer, gyro, or
+magnetometer. In these cases the driver will typically provide blocks
+encoded in the sensors native format which is often times the most compact
+representation.
+
+A sink may also be a source if it involves some processing. One common task is
+taking the raw byte layout coming from a device such as a 9-DoF intertial motion
+sensors and converting it to SI units with common numerical formats for that
+sensor kind.
+
+Potentially leading to other processing steps, perhaps to determine a gravity
```

Comment:
Delete the blank line above this so it doesn't start another paragraph and rewrite as:
```suggestion
That could then lead to other processing steps, perhaps to determine a gravity
```